src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -68,13 +69,28 @@ def drop_table(self, name: str) -> None:  # pragma: no cover
         logging.debug(f"Ready to drop {common_log_text}")
         query_id = None
         try:
-            async_job = self.session.sql(
-                f"drop table if exists {name} /* internal query to drop unused temp table */",
-            )._internal_collect_with_tag_no_telemetry(
-                block=False, statement_params={DROP_TABLE_STATEMENT_PARAM_NAME: name}
-            )
-            query_id = async_job.query_id
-            logging.debug(f"Dropping {common_log_text} with query id {query_id}")
+            if (
```

Comment:
why do we need to skip it in stored proc?

Reply:
this is equivalent to our server_connection.execute method:

https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/server_connection.py#L562-L581

Reply:
can you add this check here instead? https://github.com/snowflakedb/snowpark-python/blob/952599b1395dad9047dfc75f51324eb6014ada2a/src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py#L54, which seems cleaner

Reply:
sure, I can make the change